fbw-common/src/wasm/systems/systems/src/hydraulic/cargo_doors.rs

```diff
@@ -0,0 +1,276 @@
+    aerodynamic_model::AerodynamicModel,
+    linear_actuator::{
+        ElectroHydrostaticPowered, HydraulicAssemblyController, HydraulicLinearActuatorAssembly,
+        HydraulicLocking, LinearActuatorMode,
+    },
+    Actuator,
+};
+
+use std::time::Duration;
+
+#[derive(PartialEq, Eq, Clone, Copy)]
+pub enum DoorControlState {
+    DownLocked = 0,
+    NoControl = 1,
+    HydControl = 2,
+    UpLocked = 3,
+}
+
+pub struct HydraulicDoorController {
+    requested_position_id: VariableIdentifier,
+
+    control_state: DoorControlState,
+
+    position_requested: Ratio,
+
+    duration_in_no_control: Duration,
+    duration_in_hyd_control: Duration,
+
+    time_for_crew_to_activate_hydraulics: Duration,
+
+    should_close_valves: bool,
+    control_position_request: Ratio,
+    should_unlock: bool,
+}
+impl HydraulicDoorController {
+    // Duration which the hydraulic valves sends a open request when request is closing (this is done on real aircraft so uplock can be easily unlocked without friction)
+    const UP_CONTROL_TIME_BEFORE_DOWN_CONTROL: Duration = Duration::from_millis(200);
+
+    // Delay from the ground crew unlocking the door to the time they start requiring up movement in control panel
+    pub const DELAY_UNLOCK_TO_HYDRAULIC_CONTROL: Duration = Duration::from_secs(5);
+    const STD_DEVIATION_RAND_TIME_TO_HYD_CONTROL: Duration = Duration::from_millis(800);
+
+    pub fn new(context: &mut InitContext, id: &str) -> Self {
+        Self {
+            requested_position_id: context.get_identifier(format!("{}_DOOR_CARGO_OPEN_REQ", id)),
+            control_state: DoorControlState::DownLocked,
+            position_requested: Ratio::new::<ratio>(0.),
+
+            duration_in_no_control: Duration::from_secs(0),
+            duration_in_hyd_control: Duration::from_secs(0),
+
+            time_for_crew_to_activate_hydraulics: Self::random_hyd_control_time(),
+
+            should_close_valves: true,
+            control_position_request: Ratio::new::<ratio>(0.),
+            should_unlock: false,
+        }
+    }
+
+    fn random_hyd_control_time() -> Duration {
+        Duration::from_secs_f64(random_from_normal_distribution(
+            Self::DELAY_UNLOCK_TO_HYDRAULIC_CONTROL.as_secs_f64(),
+            Self::STD_DEVIATION_RAND_TIME_TO_HYD_CONTROL.as_secs_f64(),
+        ))
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        door: &CargoDoor,
+        current_pressure: &impl SectionPressure,
+    ) {
+        self.control_state =
+            self.determine_control_state_and_lock_action(door, current_pressure.pressure());
+        self.update_timers(context);
+        self.update_actions_from_state();
+    }
+
+    fn update_timers(&mut self, context: &UpdateContext) {
+        if self.control_state == DoorControlState::NoControl {
+            self.duration_in_no_control += context.delta();
+        } else {
+            self.duration_in_no_control = Duration::from_secs(0);
+        }
```

Comment:
Could use a simple assignment here, might look cleaner... but also looks fine as is

Reply:
What do you mean exactly?